### ISSUE TITLE
Fix accessibility issues for the @ mention list in Connect (chat)

### ISSUE THREAD

**Issue by nickytonline** [forem/forem#12017]
**Describe the bug**

Semantically, the @ mention list in Connect would make more sense as an unordered list, i.e. `<ul><li>...</li>...</ul>` and that real buttons are used instead of a `<div />` with a `role` attribute with the value `"button"`. This will help fix some issues with accessibility and it's semantically correct as it really is a list.

Other things to consider for this are ensuring that focus works properly as well as focus styles, colour contrasts etc. More info at https://docs.forem.com/frontend/accessibility/.

_Originally posted by @nickytonline in https://github.com/forem/forem/pull/10493#discussion_r547625281_

**To Reproduce**

Just see the attached screenshot to see the markup.

**Expected behavior**

**Screenshots**

![image](https://user-images.githubusercontent.com/833231/102957629-e90bef00-44a8-11eb-9a8e-a6bad370979a.png)

**Desktop (please complete the following information):**

- OS, version:
- Browser, version:

**Smartphone (please complete the following information):**

- Device:
- OS, version:
- Browser, version:

**Additional context**

**Comment by nickytonline** [comment]
Other things to consider for this are ensuring that focus works properly as well as focus styles, colour contrasts etc. More info at https://docs.forem.com/frontend/accessibility/.

**Comment by jacobherrington** [comment]
Haven't had enough time to focus on this and honestly forgot it as assigned to me. I'll come back to it once my assigned tasks are caught up (unless someone has knocked it out).

**Comment by vaidehijoshi** [comment]
@jacobherrington I'm going to add this back into "frontend focused" so that other folks know they can pick it up in the meantime :)

**Comment by citizen428** [comment]
@cmgorton Again tagged this as "ready for dev" and "good first issue" as it's very localized to one area of the codebase and well-described, so might be worth putting on the MLH board.

**Comment by cmgorton** [comment]
Thanks @citizen428. I was under the impression that the issues in the current cycle board weren't up for contributors since they are labeled Forem team. Is that the wrong assumption?

**Comment by vaidehijoshi** [comment]
Given that we have a bunch of heavyweight RFCs to either implement or flesh out this cycle and that this is well-scoped and isolated to Connect, I think we’re good to  move this to the MLH board if they have the bandwidth 👍🏽 Our team might not be able to get to it this cycle given other higher priority tasks, so I think it’s all good to move this off if we think other folks will be well-equipped to pick it up

**Comment by citizen428** [comment]
Sorry, @cmgorton, I should have added more context here, but basically what @vaidehijoshi said. If MLH has the bandwidth for this issue, I think we should remove the "Forem team" label and hand it over to them.

**Comment by nickytonline** [comment]
Since we gave the mention autocomplete feature RFC currently being implemented in #12694, we should hold off on this until that is completed so we can use that instead for this issue.

**Comment by cmgorton** [comment]
Hey @nickytonline not sure I understand what you mean. Should I remove this from the MLH project board? Are we implementing this feature in a different RFC instead?

**Comment by nickytonline** [comment]
No, it's still good for MLH to work on @cmgorton . What I meant was, we have a component for this now that @aitchiss is working on that they'll be able to use to fix this issue.

**Comment by cmgorton** [comment]
@nickytonline can you link to the PR where the component is that @aitchiss worked on.

**Comment by sammychinedu2ky** [comment]
Please can you assign me this issue

**Comment by nickytonline** [comment]
@sammychinedu2ky, you can look at the work @aitchiss did in #13126

**Comment by sammychinedu2ky** [comment]
Please do you know any reason why the user @mention auto complete won't appear on my local setup. It works for @mkcode.
We tried to figure it out unfortunately if works for him and not for me. Please is there any solution to this?

**Comment by nickytonline** [comment]
@sammychinedu2ky thanks so much for working on this! Ensure that Elastic Search is running as the mention autocomplete grabs it's data from there. See https://docs.forem.com/backend/elasticsearch/ and to set up Elastic Search and to ensure it's configured properly, check out the docs:

* [Mac](https://docs.forem.com/installation/mac/)
* [Windows](https://docs.forem.com/installation/windows/)
* [Linux](https://docs.forem.com/installation/linux/)

**Comment by sammychinedu2ky** [comment]
Thanks for your response @nickytonline , I have elastic search set up but I'm still getting the same error.

**Comment by aitchiss** [comment]
> Thanks for your response @nickytonline , I have elastic search set up but I'm still getting the same error.

Hi @sammychinedu2ky can you let us know what error you're seeing?

**Comment by sammychinedu2ky** [comment]
The automatic username box doesn't appear when I use the @ mention symbol

**Comment by aitchiss** [comment]
@sammychinedu2ky just to be extra sure can I double check that you are typing `@` plus two characters - e.g. `@s` won't trigger it, but at `@su` would? Sorry if it's an obvious question, but just to be sure! Also can I just double check where you are testing this out (e.g. typing in a comment on a post, or in the post editor)?

Are there any messages in your console? What browser/version/OS are you using? Lot's of questions! We'll get to the bottom of it 😄

**Comment by sammychinedu2ky** [comment]
Thanks 😅..
Yes I actually ensure that the characters are at least two in length but still it doesn't even make any request in the network tab ( it doesn't make the username request)

**Comment by nickytonline** [comment]
@sammychinedu2ky, as @aitchiss mentioned, if you could provide some logs from the browser console or from the terminal where you ran bin/startup, it would be helpful to debug the issues you're having. Thanks!

**Comment by sammychinedu2ky** [comment]
Sorry for the false alarm. It is working now. I wasn't using the correct repository

**Comment by aitchiss** [comment]
@sammychinedu2ky that's great to hear! thank for letting us know - and give us a shout if you have any other questions 🙂

**Comment by sammychinedu2ky** [comment]
@aitchiss  please which link leads to the channels page in this picture
https://user-images.githubusercontent.com/833231/102957629-e90bef00-44a8-11eb-9a8e-a6bad370979a.png

**Comment by aitchiss** [comment]
> @aitchiss please which link leads to the channels page in this picture
> https://user-images.githubusercontent.com/833231/102957629-e90bef00-44a8-11eb-9a8e-a6bad370979a.png

@sammychinedu2ky you can reach this page by clicking the chat icon from the home page (see below) or by visiting `/connect`. Hope this helps!

![Screenshot showing the Forem local instance and a red circle around the chat icon at the top right](https://user-images.githubusercontent.com/20773163/113585935-cb603100-9624-11eb-9ce7-fec3228d35cc.png)

**Comment by sammychinedu2ky** [comment]
Thanks a lot I am grateful

**Comment by sammychinedu2ky** [comment]
I tried using the connect feature locally but no luck. Please is there any guide line that shows how to use it?

**Comment by sammychinedu2ky** [comment]
@aitchiss  I tried using the connect feature locally but no luck. Please is there any guide line that shows how to use it?

**Comment by nickytonline** [comment]
@sammychinedu2ky,make sure you have pusher set up as that is what connect uses. See our documentation on [setting up pusher](https://docs.forem.com/backend/pusher/)

**Comment by sammychinedu2ky** [comment]
Thanks a lot @nickytonline

**Comment by sammychinedu2ky** [comment]
thanks I have it working now

**Comment by sammychinedu2ky** [comment]
Made a pull request

**Comment by sammychinedu2ky** [comment]
It can be seen here https://github.com/forem/forem/pull/13271

**Comment by cmgorton** [comment]
I am going to close this issue since we are moving to deprecate connect. 
[Deprecating Connect Chat](https://forem.dev/foremteam/deprecating-connect-chat-2kgi)
